src/lib/common/sol-platform-detect.c

```diff
@@ -266,3 +266,24 @@ sol_platform_detect(void)
     sol_file_reader_close(json_doc);
     return platform;
 }
+
+bool
+sol_platform_invalid_name(const char *name)
+{
+    unsigned int i;
+
+    if (name[0] == '\0')
+        return true;
+
+    for (i = 0; name[i] != '\0'; i++) {
+        // https://xkcd.com/276/
+        if ((name[i] >= 'A' && name[i] <= 'Z') ||
```

Comment:
Actually I will change to `isalnum` because this patch is lacking the digit check, so thanks for the catch ... and bye bye xkcd link :-(